File: 01_1.js
const fs = require('fs')
const readline = require('readline')
const events = require('events')
input_file = "01input.txt"
const data = fs.readFileSync(input_file, {encoding:'utf8', flag:'r'});

function main(ex, file){
    totals = data
        .split("\n\n")
        .map(v => v
            .split("\n")
            .filter(line => line !== "")
            .reduce((accum, value) => accum + parseInt(value), 0)
        )
        .sort((a,b) => a - b)
        .reverse()
    if (ex === 1){
        console.log(totals[0])
    }
    else{
        console.log(totals.slice(0,3).reduce((accum, value) => accum + parseInt(value), 0))
    }
}

async function doit(){
    await main(1, input_file)
    await main(2, input_file)
}

doit()

File: 01_1.py
import os
import sys
import functools

curr_dir = os.path.dirname(os.path.realpath(__file__))
input_file = f"{curr_dir}/01input.txt"

def main(ex, file):
    with open(file, "r") as f:
        elf_sums = list(
            map(
                lambda scores :
                    functools.reduce(
                        lambda a, b : a + b,
                        map(int,
                            filter(
                                lambda l : l != "",
                                scores.split("\n")
                            )
                        )
                    ),
                "".join(f).split("\n\n")
            )
        )
        elf_sums.sort()
        if ex == 1:
            print(elf_sums[-1])
        else:
            print(sum(elf_sums[-3:]))

main(1, input_file)
main(2, input_file)
